gnovm/pkg/gnomod/pkg.go

```diff
@@ -165,3 +167,94 @@ func (sp SortedPkgList) GetNonDraftPkgs() SortedPkgList {
 	}
 	return res
 }
+
+// SubPkgsFromPaths returns a list of subpackages from the given paths.
+func SubPkgsFromPaths(paths []string) ([]*SubPkg, error) {
+	for _, path := range paths {
+		fi, err := os.Stat(path)
+		if err != nil {
+			return nil, err
+		}
+		if fi.IsDir() {
+			continue
+		}
+		if filepath.Ext(path) != ".gno" {
+			return nil, fmt.Errorf("files must be .gno files: %s", path)
+		}
+
+		subPkg, err := GnoFileSubPkg(paths)
+		if err != nil {
+			return nil, err
+		}
+		return []*SubPkg{subPkg}, nil
+	}
+
```

Comment:
What do you think about extracting this out into a separate method?

I'm not sure in which case does the code on line 192 actually execute. Is it when all of the paths in the `paths` array are directories?

Reply:
Yes. When all paths are dir. 

Mixed args are not allowed. Either all args should be gno files or pkgs